SensusService/DataStores/DataStore.cs

```diff
@@ -258,10 +258,13 @@ public Task AddAsync(Datum datum, CancellationToken cancellationToken)
 
                 SensusServiceHelper.Get().Logger.Log("Stored datum:  " + datum.GetType().Name, LoggingLevel.Debug, GetType());
 
-                // if we've accumulated a chunk, commit it locally to reduce memory pressure
-                if (_data.Count >= COMMIT_CHUNK_SIZE && !_sizeTriggeredCommitRunning)
+                // if we've accumulated a chunk, commit it locally to reduce memory pressure; also commit if runLocalCommitOnStore is true
+                if ((_data.Count >= COMMIT_CHUNK_SIZE && !_sizeTriggeredCommitRunning) || runLocalCommitOnStore)
                 {
-                    SensusServiceHelper.Get().Logger.Log("Running size-triggered commit.", LoggingLevel.Normal, GetType());
+                    if (!runLocalCommitOnStore)
+                        SensusServiceHelper.Get().Logger.Log("Running size-triggered commit.", LoggingLevel.Normal, GetType());
+                    else
+                        SensusServiceHelper.Get().Logger.Log("Running Local-Commit-On-Store commit.", LoggingLevel.Normal, GetType());
 
                     _sizeTriggeredCommitRunning = true;
```

Comment:
`_sizeTriggeredCommitRunning` is no longer a good name for what is happening, since it could be either size-triggered or forced. Could also disentangle the logic a bit.